contracts/AelinPool.sol

```diff
@@ -385,6 +445,83 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+
+    function purchasePoolTokensWithNft(
+        address _collectionAddress,
+        uint256[] calldata _tokenIds,
+        uint256 _purchaseTokenAmount
+    ) external lock {
+        require(hasNftList, "pool does not have an NFT list");
+        require(block.timestamp < purchaseExpiry, "not in purchase window");
+        require(!nftWalletUsedForPurchase[msg.sender], "wallet already used");
+
+        NftData memory tmpNftData = nftCollectionDetails[_collectionAddress];
+        require(tmpNftData.collectionAddress == _collectionAddress, "collection not in the pool");
+
+        if (tmpNftData.purchaseAmountPerToken) {
+            require(
+                _purchaseTokenAmount <= tmpNftData.purchaseAmount * _tokenIds.length,
+                "purchase amount per token should be less than the allocation"
+            );
+        }
+
+        if (!tmpNftData.purchaseAmountPerToken && tmpNftData.purchaseAmount > 0) {
+            require(
+                nftAllowList[msg.sender] + _purchaseTokenAmount <= tmpNftData.purchaseAmount,
+                "purchase amount should be less than the allocation"
+            );
+            nftAllowList[msg.sender] += _purchaseTokenAmount;
+        }
+
+        if (NftCheck.supports721(_collectionAddress)) _purchasePoolWith721(_collectionAddress, _tokenIds);
+        if (NftCheck.supports1155(_collectionAddress)) _purchasePoolWith1155(_collectionAddress, _tokenIds, tmpNftData);
+        if (_collectionAddress == CRYPTO_PUNKS) _purchasePoolWithPunks(_collectionAddress, _tokenIds);
+
+        nftWalletUsedForPurchase[msg.sender] = true;
+        IERC20(purchaseToken).safeTransferFrom(msg.sender, address(this), _purchaseTokenAmount);
+        _mint(msg.sender, _purchaseTokenAmount);
+        emit PurchasePoolToken(msg.sender, _purchaseTokenAmount);
```

Comment:
technically we don't need these events since _mint calls a Transfer event from the null address that we can use instead. It was my fault for including it before but I think we can get rid of it depending on how we use the graph. I will double check tomorrow that we dont use these events. I think we can just remove it

Reply:
cool. will remove it.